tests/test_app.js

```diff
@@ -14,22 +14,33 @@
  * limitations under the License.
  *******************************************************************************/
 var global = false;
+
 process.argv.forEach(function(elem) {
-  if (elem == '-g')
+  if (elem == '-g'){
     global = true;
+  }
 });
 
 var agent;
-if (!global)
-{
-  agent = require('../');
-  agent.start();
 
-  // Make agent visible for other script files.
-  module.exports.agent = agent;
+//If running global test, run long enough to ensure the agent has loaded and process doesn't crash
+if (global) {
+	var duration_secs = process.argv[2] || 10; //Default 10 seconds for global tests
```

Comment:
Is there a way to test if the agent has started up and remove the wait 10 seconds?
(Otherwise this is likely to fail randomly on slower machines!)